Adding security installation information #20231130 (#6605)
opensearch-project/documentation-website: _security/configuration/security-installation.md

---
layout: default

> title: Security installation

Reviewer: This file should be deleted. Just keep the 'Enabling security' file.

---
layout: default

> title: Security installation

Reviewer: Enabling security

Reviewer: See comment below...

Author: I think this was regarding the name change, thats updated now

parent: Configuration

> nav_order: 60

Reviewer: ```suggestion
nav_order: 45
```

---

> # Security installation

Author: Great idea, updated now

> Default version of OpenSearch comes with Security features pre-installed, however if this was [disabled]({{site.url}}{{site.baseurl}}/security/configuration/disable/) or OpenSearch Minimum was installed, this page goes through the steps of installing Security features.

Reviewer: We don't ever refer to the minimal distribution version of OpenSearch "OpenSearch Minimum" in the documentation. Perhaps we can say, ....if the Security plugin was disabled (link) or OpenSearch was installed without security using the minimal distribution method..."

Author: Updated as per suggestion

> Default version of OpenSearch comes with Security features pre-installed, however if this was [disabled]({{site.url}}{{site.baseurl}}/security/configuration/disable/) or OpenSearch Minimum was installed, this page goes through the steps of installing Security features.

Reviewer: ```suggestion
The default version of OpenSearch comes with Security features pre-installed, however if the Security plugin was [disabled]({{site.url}}{{site.baseurl}}/security/configuration/disable/) or OpenSearch was installed without security, for example, using the minimal distribution method, you can enable the plugin as follows. 
```

> Full cluster restart is necessary to enable security features

Reviewer: ```suggestion
A full cluster restart is necessary to enable security features.
```

{: .label .label-red }

## Installation steps for prod cluster

> 1. Disable shard allocation and stop all nodes

Reviewer: ```suggestion
1. Disable shard allocation and stop all nodes in order to prevent shards from moving around when the cluster is restarted.
```

>   You need to disable in order to prevent shards moving around when the cluster is restarted.

Reviewer: ```suggestion
```

```json
  PUT "http://localhost:9200/_cluster/settings" -H 'Content-Type: application/json' -d '{
  "transient": {
    "cluster.routing.allocation.enable": "none"
  }
}'
```
{% include copy-curl.html %}

> 2. Install plugin on all nodes

Reviewer: ```suggestion
2. Install the plugin on all nodes.
```

```bash
bin/opensearch-plugin install opensearch-security
```
{% include copy.html %}

> 3. Add necessary configuration to opensearch.yml for TLS encryption

Reviewer: ```suggestion
3. Add the necessary configuration to opensearch.yml for TLS encryption.
```

[Configuration]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/security-settings/) details different settings which need to be configured

> 4. Create OPENSEARCH_INITIAL_ADMIN_PASSWORD variable

Reviewer: ```suggestion
4. Create the OPENSEARCH_INITIAL_ADMIN_PASSWORD variable.
```

> 5. Restart the nodes and reenable shard allocation

Reviewer: ```suggestion
5. Restart the nodes and reenable shard allocation.
```
